drivers/soundwire/stream.c

```diff
@@ -1665,12 +1668,24 @@ static int _sdw_deprepare_stream(struct sdw_stream_runtime *stream)
 			return ret;
 		}
 
+		multi_lane_bandwidth = 0;
+
+		list_for_each_entry(p_rt, &m_rt->port_list, port_node) {
+			if (!p_rt->lane)
+				continue;
+
+			bandwidth = m_rt->stream->params.rate * hweight32(p_rt->ch_mask) *
+				m_rt->stream->params.bps;
+			multi_lane_bandwidth += bandwidth;
+			bus->lane_used_bandwidth[p_rt->lane] -= bandwidth;
+		}
 		/* TODO: Update this during Device-Device support */
-		bus->params.bandwidth -= m_rt->stream->params.rate *
-			m_rt->ch_count * m_rt->stream->params.bps;
+		bandwidth = m_rt->stream->params.rate * m_rt->ch_count * m_rt->stream->params.bps;
+		bus->params.bandwidth -= bandwidth - multi_lane_bandwidth;
```

Comment:
this should be fixed here instead of having a separate patch "soundwire: subtract bandwidth after compute_params in deprepare_stream" which has problems too.

Reply:
Don't you think they are 2 different issues? The needed of moving the bus->params.bandwidth change after bus->compute_params() is not because we added multi-lane support. The issue exists today. We didn't meet the issue because we use a fixed clock rate. The issue only happens when the 4th stream is closed. The scenario is that we use a higher clock and new frame shape when the 4th stream is opened. And when any stream is closed, i.e. the active stream number become 3, sdw_compute_bus_params() will select lower clock rate and the original frame shape. But, sdw_compute_group_params() still calculate the `payload_bw` for 4 streams since the m_rt is not removed yet. Then sdw_compute_group_params() will return -EINVAL because `column_needed > sel_col - 1`.

Reply:
sure, then move this patch earlier